src/uu/dd/src/dd.rs

```diff
@@ -0,0 +1,1296 @@
+        return Ok((rstat, buf));
+    }
+
+    // Perform any conv=x[,x...] options
+    if i.cflags.swab {
+        perform_swab(&mut buf);
+    }
+    if is_conv(i) || is_block(i) || is_unblock(i) {
+        let buf = conv_block_unblock_helper(buf, i, &mut rstat)?;
+        Ok((rstat, buf))
+    } else {
+        Ok((rstat, buf))
+    }
+}
+
+fn print_io_lines(update: &ProgUpdate) {
+    eprintln!(
+        "{}+{} records in",
+        update.reads_complete, update.reads_partial
+    );
+    if update.records_truncated > 0 {
+        eprintln!("{} truncated records", update.records_truncated);
+    }
+    eprintln!(
+        "{}+{} records out",
+        update.writes_complete, update.writes_partial
+    );
+}
+fn make_prog_line(update: &ProgUpdate) -> String {
+    let btotal_metric = Byte::from_bytes(update.bytes_total)
+        .get_appropriate_unit(false)
+        .format(0);
+    let btotal_bin = Byte::from_bytes(update.bytes_total)
+        .get_appropriate_unit(true)
+        .format(0);
+    let safe_millis = cmp::max(1, update.duration.as_millis());
+    let xfer_rate = Byte::from_bytes(1000 * (update.bytes_total / safe_millis))
+        .get_appropriate_unit(false)
+        .format(1);
+
+    format!(
+        "{} bytes ({}, {}) copied, {:.1} s, {}/s",
+        update.bytes_total,
+        btotal_metric,
+        btotal_bin,
+        update.duration.as_secs_f64(),
+        xfer_rate
+    )
+}
+fn reprint_prog_line(update: &ProgUpdate) {
+    eprint!("\r{}", make_prog_line(update));
+}
+fn print_prog_line(update: &ProgUpdate) {
+    eprintln!("{}", make_prog_line(update));
+}
+fn print_xfer_stats(update: &ProgUpdate) {
+    print_io_lines(update);
+    print_prog_line(update);
+}
+
+/// Generate a progress updater that tracks progress, receives updates, and responds to signals.
+fn gen_prog_updater(rx: mpsc::Receiver<ProgUpdate>, xfer_stats: Option<StatusLevel>) -> impl Fn() {
+    // --------------------------------------------------------------
+    fn posixly_correct() -> bool {
+        env::var("POSIXLY_CORRECT").is_ok()
+    }
+    // --------------------------------------------------------------
+    move || {
+        const SIGUSR1_USIZE: usize = signal::SIGUSR1 as usize;
+
+        let sigval = Arc::new(AtomicUsize::new(0));
+
+        // TODO: SIGINFO seems to only exist for BSD (and therefore MACOS)
+        // I will probably want put this behind a feature-gate and may need to pass the value to handle as my own constant.
+        // This may involve some finagling with the signals library.
+        // see -> https://unix.stackexchange.com/questions/179481/siginfo-on-gnu-linux-arch-linux-missing
+        // if let Err(e) = signal_hook::flag::register_usize(signal::SIGINFO, sigval.clone(), signal::SIGINFO as usize)
+        // {
+        //     debug_println!("Internal dd Warning: Unable to register SIGINFO handler \n\t{}", e);
+        // }
+        if !posixly_correct() {
+            if let Err(e) =
```

Comment:
I'm not sure about this one. Changing the variable name, (and presumably also removing the print statement below it) means a silent error if the signal handler fails to register.

Or maybe you meant to change a different line (I think 794?).

Reply:
Oh maybe, I meant the one that caused the unused variable warning

Reply:
The warning is because the print is only there in debug mode, so in release mode `e` is unused. Maybe we want to print the warning in release mode too? Otherwise I think it's fine to ignore this (add `#[allow(unused_variables)]`)

Reply:
I think miDeb is right about the issue.

I've adjusted this to print the warning on all builds unless `status=none` is specified.